problems/P047/assignment-47.py

```diff
@@ -0,0 +1,20 @@
+from bitarray import bitarray
+import datetime
+x=int(input("Enter a number:"))
+s=datetime.datetime.now()
+prime=bitarray(x+1)
+prime.setall(True)
+i=3
+while i*i<x+1:
+    if prime[i]==True:
+        for j in range(i*i,x+1,2*i):
+            prime[j]=False
```

Comment:
Add a single if condition and check prim[j] before assigning False to it.

Count how many times you are re-assigning False to all nodes?

All of those re-assignments can be theorically be eliminiated, practically a portion of them. The more you eliminate the less operation you do, the faster your sieve will be

Reply:
I added if condition as you'd mentioned but the time in problem 54 increased exactly 20 percent